staging/src/k8s.io/apiserver/pkg/server/signal.go

```diff
@@ -44,6 +45,18 @@ func SetupSignalHandler() <-chan struct{} {
 	return stop
 }
 
+// SetupSignalHandlerContext observe SIGTERM and SIGINIT and cancel the passed context
+// when one of the signals is caught.
+// Can only be called once and cannot be called if SetupSignalHandler is already used.
+func SetupSignalHandlerContext(ctx context.Context) {
```

Comment:
don't like the signature.

It should rather look like the Golang context wrapper: https://golang.org/pkg/context/#WithTimeout

Reply:
WithSignalHandler(ctx) (ctx, cancel)

Reply:
yeah, I like the WithSignals better, fixed.